src/View/Form/ArrayContext.php

```diff
@@ -212,19 +213,20 @@ public function isRequired($field)
      * In this context class, this is simply defined by the 'maxlength' array.
      *
      * @param string $field A dot separated path to check required-ness for.
-     * @return int|false
+     * @return int|null
      */
     public function getMaxLength($field)
     {
         if (!is_array($this->_context['schema'])) {
-            return false;
+            return null;
         }
         $fieldShema = Hash::get($this->_context['schema'], $field);
-        if(is_array($fieldShema)) {
+
+        if (is_array($fieldShema)) {
             $maxLength = Hash::get($fieldShema, 'length');
         }
 
-        return $maxLength ?? false;
+        return $maxLength ?? null;
```

Comment:
isnt $maxLength already nullable? The cloaking ?? should be avoided here then.

Reply:
Actually, since those are existing API methods, better to keep false, see below.